server/src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -291,13 +292,11 @@ public <T extends TransportRequest> RequestHandlerRegistry<T> getHandler(String
             return (RequestHandlerRegistry<T>) requestHandlers.get(action);
         }
 
-        public List<TransportActionStats> getStats() {
+        public Map<String, TransportActionStats> getStats() {
             return requestHandlers.values()
                 .stream()
                 .filter(reg -> reg.getStats().requestCount() > 0 || reg.getStats().responseCount() > 0)
-                .sorted(Comparator.comparing(RequestHandlerRegistry::getAction))
-                .map(RequestHandlerRegistry::getStats)
-                .toList();
+                .collect(Collectors.toMap(RequestHandlerRegistry::getAction, RequestHandlerRegistry::getStats, (a, b) -> a, TreeMap::new));
```

Comment:
Re `(a, b) -> a`, do we expect a duplicate action somewhere?

Reply:
no, but we've got to say _something_ here (there's no overload that accepts a `mapFactory` without a `mergeFunction`) and this is the shortest thing we can say

Reply:
Oh, okay. NIT: we have `org.elasticsearch.common.util.Maps#toUnmodifiableSortedMap` that only accept key/value functions and could be used here.